java/src/client/data/PlayerInfo.java

```diff
@@ -27,6 +31,17 @@ public PlayerInfo() {
 		setName("");
 		setColor(CatanColor.WHITE);
 	}
+
+    //added this function
+	public PlayerInfo(String n, int vp, CatanColor c, int i, int index, boolean lr, boolean la){
```

Comment:
Add a few asserts here, things like
n != null
c != null
i >= 0
etc.

Also could the variable names be more informative? In IntelliJ when you begin using the constructor, it'll show you the parameter names, but if I'm using this class I'll have no idea what "n", "vp", "c" etc. is.